src/main/java/org/HdrHistogram/StripedWriterReaderPhaser.java

```diff
@@ -0,0 +1,227 @@
+            oddEndEpoch.incrementAndGet(threadIndex());
+        } else {
+            evenEndEpoch.incrementAndGet(threadIndex());
+        }
+    }
+
+    /**
+     * Enter to a critical section containing a read operation (mutually excludes against other
+     * {@link StripedWriterReaderPhaser#readerLock} calls).
+     * <p>
+     * {@link StripedWriterReaderPhaser#readerLock} DOES NOT provide synchronization
+     * against {@link StripedWriterReaderPhaser#writerCriticalSectionEnter()} calls. Use {@link StripedWriterReaderPhaser#flipPhase()}
+     * to synchronize reads against writers.
+     */
+    public void readerLock() {
+        readerLock.lock();
+    }
+
+    /**
+     * Exit from a critical section containing a read operation (relinquishes mutual exclusion against other
+     * {@link StripedWriterReaderPhaser#readerLock} calls).
+     */
+    public void readerUnlock() {
+        readerLock.unlock();
+    }
+
+    /**
+     * Flip a phase in the {@link StripedWriterReaderPhaser} instance, {@link StripedWriterReaderPhaser#flipPhase()}
+     * can only be called while holding the readerLock().
+     * {@link StripedWriterReaderPhaser#flipPhase()} will return only after all writer critical sections (protected by
+     * {@link StripedWriterReaderPhaser#writerCriticalSectionEnter()} ()} and
+     * {@link StripedWriterReaderPhaser#writerCriticalSectionExit(long)} ()}) that may have been in flight when the
+     * {@link StripedWriterReaderPhaser#flipPhase()} call were made had completed.
+     * <p>
+     * No actual writer critical section activity is required for {@link StripedWriterReaderPhaser#flipPhase()} to
+     * succeed.
+     * <p>
+     * However, {@link StripedWriterReaderPhaser#flipPhase()} is lock-free with respect to calls to
+     * {@link StripedWriterReaderPhaser#writerCriticalSectionEnter()} and
+     * {@link StripedWriterReaderPhaser#writerCriticalSectionExit(long)}. It may spin-wait for for active
+     * writer critical section code to complete.
+     *
+     * @param yieldTimeNsec The amount of time (in nanoseconds) to sleep in each yield if yield loop is needed.
+     */
+    public void flipPhase(long yieldTimeNsec) {
+        if (!readerLock.isHeldByCurrentThread()) {
+            throw new IllegalStateException("flipPhase() can only be called while holding the readerLock()");
+        }
+
+        boolean nextPhaseIsEven = (startEpoch.get(0) < 0); // Current phase is odd...
+
+        long initialStartValue;
+        // First, clear currently unused [next] phase end epoch (to proper initial value for phase):
+        if (nextPhaseIsEven) {
+            initialStartValue = 0;
+            for (int i=0; i<stripes; i++) {
+                evenEndEpoch.lazySet(i, initialStartValue + i * stride);
+            }
+        } else {
+            initialStartValue = Long.MIN_VALUE;
+            for (int i=0; i<stripes; i++) {
+                oddEndEpoch.lazySet(i, initialStartValue + i * stride);
```

Comment:
We don’t really need he epoch start values to use a different range per stripe. All stripes can independently start at either 0 (fir even) or Long.Min_VALUE (for odd). So we can remove the + i * stride logic here and in the initialization above.

Reply:
Removed the strides logic.